File: Code Samples/OOP/OOP II - Harsh Duche/Java/METHOD_OVERRIDING.java
class gdsc_member {
    String name;

    public gdsc_member(String name) {
        this.name = name;
    }

    public void introduce() {
        System.out.println("My name is " + this.name + " & I'm a GDSC member");
    }
}

class dev_team extends gdsc_member {
    String domain;

    public dev_team(String name, String domain) {
        super(name);
        this.domain = domain;
    }

    public void introduce() {
        System.out.println("My name is " + this.name + " & I'm a development team member. My domain is " + this.domain);
    }
}

public class METHOD_OVERRIDING {
    public static void main(String[] args) {
        gdsc_member member1 = new gdsc_member("Harsh");
        dev_team dev1 = new dev_team("Harsh", "AI/ML");

        member1.introduce();
        dev1.introduce();
    }
}

File: Code Samples/OOP/OOP II - Harsh Duche/Python/METHOD OVERRIDING.py
class gdsc_member:
    def __init__(self, name):
        self.name = name

    def introduce(self):
        print(f"My name is {self.name} & I'm a GDSC member")

class dev_team(gdsc_member):
    def __init__(self, name, domain):
        super().__init__(name)
        self.domain = domain

    def introduce(self):
        print(f"My name is {self.name} & I'm a development team member. My domain is {self.domain}")


member1 = gdsc_member("Harsh")
dev1 = dev_team("Harsh","AI/ML")

member1.introduce()
dev1.introduce()